tasks/vhost.yml

```diff
@@ -2,9 +2,43 @@
 
 - name: Add apache vhosts configurations
   template:
-    src=vhost.conf.j2
-    dest=/etc/apache2/sites-available/{{ item.servername }}.conf
+    src="{{item.template | default('vhost.conf.j2')}}"
+    dest="/etc/apache2/sites-available/{{ item.server_file_name }}.conf"
     owner=root group=root mode=644
   notify: Restart apache
   with_items: apache_vhosts
   when: apache_vhosts != false
+
+- file: path="{{apache_ssl_path}}" state=directory
+
+- name: Ensure document root exists
+  file: path="{{item.document_root}}" state=directory
+  notify: Restart apache
+  with_items: apache_vhosts
+  when: apache_vhosts != false
+
+- name: Copy key
+  command: cp "{{apache_ssl_key_store}}/{{item.1.key_file}}" "{{apache_ssl_path}}"
```

Comment:
I don't use the copy module because that doesn't do remote -> remote copies.  It copies local -> remote.  This could be done, but then we have to copy the private key locally, only to copy it back again.  I would rather not do this method.

We could use the synchronize module, however that requires rsync to be installed on the remote, but it is idempotent.  This [Ansible issue](https://github.com/ansible/ansible/issues/9982) suggests using synchronize for remote -> remote copies so I'll switch to that.